src/settings.rs

```diff
@@ -36,7 +36,265 @@ pub struct FacetingSettings {
+///   model: "nomic-embed-text".to_string(),
+///   document_template: Some("A document titled {{doc.title}} whose description starts with {{doc.overview|truncatewords: 20}}".to_string()),
+/// };
+/// # let expected = r#"{"url":"http://localhost:11434/api/embeddings","apiKey":"foobarbaz","model":"nomic-embed-text","documentTemplate":"A document titled {{doc.title}} whose description starts with {{doc.overview|truncatewords: 20}}"}"#;
+/// # let expected: OllamaEmbedderSettings = serde_json::from_str(expected).unwrap();
+/// # assert_eq!(embedder_setting, expected);
+/// ```
+#[derive(Serialize, Deserialize, Default, Debug, Clone, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct OllamaEmbedderSettings {
+    /// Mandatory, full URL to the embedding endpoint.
+    /// Must be parseable as a URL.
+    /// If not specified, [Meilisearch](https://www.meilisearch.com/) (**not the sdk you are currently using**) will try to fetch the `MEILI_OLLAMA_URL` environment variable
+    /// Example: `"http://localhost:11434/api/embeddings"`
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub url: Option<String>,
+    /// Optional, token used to authenticate against [Ollama](https://ollama.com/)
+    /// Example: `"foobarbaz"`
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub api_key: Option<String>,
+    /// See https://ollama.com/library?q=embed for suitable embedding models
+    ///
+    /// # Example embedding models
+    ///
+    /// | Model                    | Parameter    | Size                                                            |
+    /// |--------------------------|--------------|-----------------------------------------------------------------|
+    /// | `mxbai-embed-large`      | `334M`       | [View model](https://ollama.com/library/mxbai-embed-large)      |
+    /// | `nomic-embed-text`       | `137M`       | [View model](https://ollama.com/library/nomic-embed-text)       |
+    /// | `all-minilm`             | `23M`,`33M`  | [View model](https://ollama.com/library/all-minilm)             |
+    /// | `snowflake-arctic-embed` | varies       | [View model](https://ollama.com/library/snowflake-arctic-embed) |
+    pub model: String,
+    /// Use it to customize the data you send to the embedder. It is highly recommended you configure a custom template for your documents.
+    ///
+    /// if present, `document_template` must be a [Liquid template](https://shopify.github.io/liquid/).
+    /// Use `{{ doc.attribute }}` to access document field values.
+    /// Meilisearch also exposes a `{{ fields }}` array containing one object per document field, which you may access with `{{ field.name }}` and `{{ field.value }}`.
+    ///
+    /// For best results, use short strings indicating the type of document in that index, only include highly relevant document fields, and truncate long fields.
+    /// Example: "A document titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}"
```

Comment:
```suggestion
    /// Example: `"A document titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}"`
    /// 
    /// Default:
    /// ```raw
    /// {% for field in fields %}
    /// {% if field.is_searchable and not field.value == nil %}
    /// {{ field.name }}: {{ field.value }}\n
    /// {% endif %}
    /// {% endfor %}
    /// ```
```